server/strapi-settings.ts

```diff
@@ -1,45 +1,34 @@
 import { writeFileSync } from "fs";
-import { getNavData, getBannerData } from "./strapi";
+import { fetchData } from "./strapi";
 
-export type GenerateEventProps = {
-  file: string; // path to file
-};
-
-export const generateEvent = async ({ file }: GenerateEventProps) => {
-  const writeEventData = (data: object) => {
-    try {
-      writeFileSync(file, JSON.stringify(data));
-      console.log("Writing event data to file...", "\n");
-      console.log("FILE:", file, "\n");
-    } catch (error) {
-      console.error("Error writing event data to file:", error);
-    }
-  };
-  let eventData = undefined;
-  eventData = await getBannerData();
-  if (eventData) {
-    writeEventData(eventData);
-  } else {
-    writeEventData({});
-  }
-};
-
-export const generateNavigation = async ({ file }: { file: string }) => {
-  let navData = undefined;
+export const generateData = async ({
+  navPath,
+  eventPath,
+}: {
+  navPath: string;
+  eventPath: string;
+}) => {
+  let data = undefined;
   try {
-    navData = await getNavData();
-    if (!navData) return;
+    data = await fetchData();
+    if (!data) return;
   } catch (error) {
-    console.error("No header navigation data returned");
+    console.error("No data returned");
     return;
   }
-  if (navData) {
+
+  if (data) {
+    try {
+      writeFileSync(navPath, JSON.stringify(data.navbardata));
```

Comment:
Can we guarantee that the data has the `navbardata` field? (Same with the `eventsdata` field)